### PR TITLE
Implement OpenAI server

### PR DESCRIPTION
**Pull request by AndyZe**
`ros2 run ros2_openai_server openai_server`

- Test sending of a string prompt with an empty image, receiving back a bool:

`ros2 service call /openai_server ai_msgs/srv/StringImagePrompt prompt:\ "are you a pirate? please respond with a one-word answer, yes or no"`

- Run the example client, which sends a picture of a wooden table and asks "Is this a wooden table?"

`ros2 run ros2_openai_server example_client`

### CODE REVIEW

**Comment by AndyZe** [comment]
This should be ready to review now  :+1:

**Comment by brettpac** [comment]
The CI error related to CURL isn't affecting you when you build on your local machine right?

**Comment by AndyZe** [comment]
> The CI error related to CURL isn't affecting you when you build on your local machine right?

Good call, need to add it to the installation instructions, I guess. Will try to fix CI now.

File: include/ros2_openai_server/convert_image_to_base_64.hpp
Comment (brettpac) [inline include/ros2_openai_server/convert_image_to_base_64.hpp:1]: This was a nice fix for the image conversion problem.

**Review by brettpac — APPROVED** [review]

**Pull request merged**